nav2_map_server/src/map_saver/main_cli.cpp

```diff
@@ -63,20 +72,31 @@ typedef enum
   HELP_MESSAGE
 } ARGUMENTS_STATUS;
 
+struct SaveParamList
+{
+  // 2D parameters
+  map_2d::SaveParameters save_parameters_2d;
+  // 3D parameters
+  map_3d::SaveParameters save_parameters_3d;
+  std::string origin_topic;
```

Comment:
What's the workflow for existing 3D SLAMs? They have a separate topic containing the origin? 

Isn't an origin in a 3D map completely unnecessary? You know what the origin is if you have a pointcloud of 3D points relative to a coordinate frame. If you have 3 points `(A, B, C)`, you have embedded in that a reference frame where the zero position is where `(0,0,0)` is. The 2D map has an origin because its in pixel non-real-world coordinates. @AlexeyMerzlyakov am I wrong? 

If I am wrong and we do require an origin, we should really be making a new message format for a 3D map that contains the pointcloud points as well as the origin so that we dont have a workflow with the separate topic -- that's not good that its not all together like the OccGrid message it.

Reply:
Was fixed in latest commit.